legacy/expressions/migrate.go

```diff
@@ -115,7 +115,6 @@ func migrateExpression(env utils.Environment, resolver types.XValue, expression
 	p.AddErrorListener(errListener)
 
 	// speed up parsing
-	p.SetErrorHandler(antlr.NewBailErrorStrategy())
```

Comment:
not sure if its a problem in ANTLR, but this causes panics when there are syntax errors

Reply:
Did you benchmark this at all? Wasn't this the thing causing crazy performance degradation. We can catch the panic it if comes to that.